assets/app/view/game/part/track_node_path.rb

```diff
@@ -411,7 +428,19 @@ def render_part
             'stroke-dasharray': @dash,
           ) if @terminal
 
-          h(:path, props)
+          children = []
+          children.append(h(:path, props))
+          if @crossover_dash
+            intersect_props = props.dup
```

Comment:
does this dup actually do what you want? dup is usually just a shallow copy, so nested dictionaries would still be mutated?

Reply:
It seems to work, but I can change to a a deep copy.

Reply:
maybe separate out the hashes

Reply:
Actually I don't need to do either. It seems each call of h(:path, props) is essentially a call by value of the properties, so I'm free to change props between calls.